### PR TITLE
boinc: Add version 7.14.2

### PR DESCRIPTION
**Pull request by jfastnacht**
I am unsure whether this is a proper way to install it, but it works. Feedback would be highly appreciated.

### CODE REVIEW

File: bucket/boinc.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+    "version": "7.14.2",
+    "description": "Open-source software for volunteer and grid computing",
```
Comment (Ash258) [inline bucket/boinc.json]:
```suggestion
    "description": "Volunteer scientific or grid computing client",
```

File: bucket/boinc.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            "hash": "a8dfa47c58dcf54d29e77434c1fb5a0e7bfdea66b5f5c128a91104449c263987",
+            "installer": {
+                "script": [
+                    "isx \"$dir\\$fname\"",
+                    "$isxextractdir = ((gci \"$dir\" | ? { $_.PSIsContainer } | sort CreationTime)[-1]).Name",
+                    "Expand-MsiArchive \"$dir\\$isxextractdir\\BOINC.msi\" \"$dir\" -ExtractDir 'Program Files 64\\BOINC'",
+                    "Remove-Item -LiteralPath \"$dir\\$fname\" -Force",
+                    "Remove-Item -LiteralPath \"$dir\\$isxextractdir\" -Force -Recurse"
+                ]
+            }
```
Comment (Ash258) [inline bucket/boinc.json]:
```suggestion
            "hash": "a8dfa47c58dcf54d29e77434c1fb5a0e7bfdea66b5f5c128a91104449c263987"
        },
        "32bit": {
            "url": "https://boinc.berkeley.edu/dl/boinc_7.14.2_windows_intelx86.exe",
            "hash": "0f79946d1501d98619f80565b21896b0e0d9732485a4a030b5c14322426aeb79"
```

File: bucket/boinc.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    },
```
Comment (Ash258) [inline bucket/boinc.json]:
```suggestion
    },
    "installer": {
        "script": [
            "Invoke-ExternalCommand ((Get-Command 'isx' -CommandType Application).Source) -ArgumentList \"$dir\\$fname\" -LogPath \"$dir\\isx.log\" | Out-Null",
            "$folder = Get-ChildItem \"$dir\" -Directory | Select-Object -First 1 -ExpandProperty FullName",
            "$extract = if ($architecture -eq '64bit') { 'Program Files 64' } else { 'program files' }",
            "Expand-MsiArchive \"$folder\\BOINC.msi\" \"$dir\" -ExtractDir \"$extract\\BOINC\" -Removal",
            "Remove-Item $folder -Force -Recurse"
        ]
    },
```

File: bucket/boinc.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            "BOINC Manager"
+        ],
+        [
+            "boincscr.exe",
+            "BOINC Screensaver"
+        ]
+    ],
+    "checkver": {
+        "url": "https://boinc.berkeley.edu/download_all.php?xml=1",
+        "xpath": "/versions/version[contains(.,'windows_x86_64')]/version_num"
```
Comment (Ash258) [inline bucket/boinc.json]:
```suggestion
        "xpath": "/versions/version[platform='Windows 64-bit']/version_num"
```

File: bucket/boinc.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://boinc.berkeley.edu/dl/boinc_$version_windows_x86_64.exe"
+            }
```
Comment (Ash258) [inline bucket/boinc.json]:
```suggestion
            },
            "32bit": {
                "url": "https://boinc.berkeley.edu/dl/boinc_$version_windows_intelx86.exe"
            }
```

**Comment by Ash258** [comment]
- Where could be configuration location set?
- What files/folders need to be persisted?

**Comment by jfastnacht** [comment]
If you do not specify a directory, the configuration files will be created in the folder you start the client in, except for running the BOINC Manager GUI, which starts a client (when none was detected) in the background, which creates the files in the installation directory. And it's unfortunately just a bunch of XML-files with names depending on the specific project and no subdirectory. I try to figure out if there is some better way to handle that.

**Comment by jfastnacht** [comment]
I am not 100% happy with this solution, since it needs administrative privileges. The problem is Boinc itself, which has no fallback to the HKEY_CURRENT_USER path to fetch INSTALLDIR and DATADIR from Boincs registry entries.

**Comment by Ash258** [comment]
I am not going to merge this with current state. If you want it with this open PR into other more .

If the applicaiton requires local machine registry to configure simple data location it is really not suitable for official bucket.

If you know what all folders and file should be persisted then it can be copied on installation/uninstallation. Otherwise it will not be in official bucket.

**Comment by jfastnacht** [comment]
I try to reach out to the Boinc developers and might reinitiate the PR afterwards.

**Pull request closed without merging**